### PR TITLE
[PAPP-32422] Tanium Threat Response: Bugfix - Bug fix for accessing path to the vault temp directory

### PR DESCRIPTION
**Pull request by bbielinski-splunk**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Bug fix for accessing path to the vault temp directory [PAPP-32422]

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: taniumthreatresponse_connector.py [hunk 1]
```diff
@@ -376,21 +376,17 @@ def _get_filename_from_tanium(self, action_result, file_id):
         return RetVal(phantom.APP_SUCCESS, filename)
 
     def _save_temp_file(self, content):
+        vault_tmp_dir = Vault.get_vault_tmp_dir()
```
Comment (bb-splunk) [inline taniumthreatresponse_connector.py:379]: Have you tried with the `add` API?

Comment (bbielinski-splunk) [reply taniumthreatresponse_connector.py:379]: The `vault_add()` method is used [a bit later ](https://github.com/splunk-soar-connectors/taniumthreatresponse/blob/6d2157c16d2b9a9c02b8f9a6e3610db1282ce888/taniumthreatresponse_connector.py#L1180C8-L1180C8). The `_save_temp_file()` is used to extract a content of tanium thread response from a zip file to unpacked file, which is then added to vault.

File: taniumthreatresponse_connector.py [hunk 2]
```diff
@@ -376,21 +376,17 @@ def _get_filename_from_tanium(self, action_result, file_id):
...
-        temp_dir = '{}/{}'.format(temp_dir, uuid.uuid4())
         os.makedirs(temp_dir)
 
         # We are getting application/zip object from tanium and it has set default password.
         # So, we are extracting zip object and store it into vault
         with ZipFile(BytesIO(content)) as zobj:
             zobj.extractall(path=temp_dir, pwd=b"infected")
 
-        return temp_dir + '/' + zobj.namelist()[0]
+        return os.path.join(temp_dir, str(zobj.namelist()[0]))
```
Comment (bb-splunk) [inline taniumthreatresponse_connector.py:389]: Is the cast to `str` necessary?

**Review by bb-splunk — APPROVED** [review]

File: taniumthreatresponse_connector.py (hunk 2 above)
Comment (bbielinski-splunk) [reply taniumthreatresponse_connector.py:389]: Yes, for the proper type checking purpose. `zobj.namelist()[0]` is not `str` so `os.path.join` raises warnings that only strings should be passed

**Pull request merged**